scripts/rv_expression_association/get_rv_outliers.py

```diff
@@ -0,0 +1,197 @@
+
+    # file matching OneK1K IDs to CPG (internal) and TOB (external) IDs
+    sample_key_df = pd.read_csv(
+        AnyPath(dataset_path(sample_mapping_file)), sep='\t', index_col=0
+    )
+
+    cpg_id = sample_key_df[sample_key_df.index == onek1k_id]['InternalID'].values[0]
+    logging.info(f'CPG ID: {cpg_id}')  # e.g., 'CPG9951'
+
+    # define output filename and check if it already exists
+    output_filename = AnyPath(output_path(f'{cpg_id}_{gene_name}_{window_size}.csv'))
+    logging.info(f'Output file: {output_filename}')
+
+    # skip if file already exists
+    if output_filename.exists():
+        raise Exception(f'File {output_filename} already exists, exiting')
+
+    init_batch()
+    # get VEP-annotated WGS object (hail matrix table)
+    # mt = hl.read_matrix_table('gs://cpg-tob-wgs-test/tob_wgs_vep/v1/vep105_GRCh38.mt')
+    mt = hl.read_matrix_table(MT)
+    logging.info(f'Number of total variants: {mt.count()[0]}')
+
+    # get relevant chromosome
+    gene_df = pd.read_csv(AnyPath(dataset_path(gene_file)), sep='\t', index_col=0)
+    chrom = gene_df[gene_df['gene_name'] == gene_name]['chr']
+
+    # filter to relevant chromosome to speed densification up
+    mt = mt.filter_rows(mt.locus.contig == ('chr'+chrom))  # figure out actual syntax
+    logging.info(f'Number of variants on chromosome {chrom}: {mt.count()[0]}')
+
+    # densify
+    mt = hl.experimental.densify(mt)
+
+    # filter out low QC variants
+    mt = mt.filter_rows(hl.len(hl.or_else(mt.filters, hl.empty_set(hl.tstr))) == 0)
+    mt_path = output_path('densified_chrom_and_qc_filtered.mt', 'tmp')
+    mt = mt.checkpoint(mt_path, overwrite=True)
+    logging.info(f'Number of QC-passing variants: {mt.count()[0]}')
+
+    # select matrix down to that one donor
+    donor_mt = mt.filter_cols(mt.s == cpg_id)
+
+    # get gene body position (start and end) and build interval
+    # include variants up to {window size} up- and downstream
+    interval_start = int(gene_df[gene_df['gene_name'] == gene_name]['start']) - int(
+        window_size
+    )
+    interval_end = int(gene_df[gene_df['gene_name'] == gene_name]['end']) + int(
+        window_size
+    )
+
+    # get gene-specific genomic interval
+    gene_interval = 'chr' + chrom + ':' + str(interval_start) + '-' + str(interval_end)
+    logging.info(f'Interval considered: {gene_interval}')  # 'chr22:23219960-23348287'
+
+    donor_mt = hl.filter_intervals(
+        donor_mt, [hl.parse_locus_interval(gene_interval, reference_genome='GRCh38')]
+    )
+    logging.info(f'Number of variants within interval: {donor_mt.count()[0]}')
+
+    # remove variants for which this individual is 0/0
+    donor_mt = hl.variant_qc(donor_mt)
+    donor_mt = donor_mt.filter_rows(donor_mt.variant_qc.n_non_ref > 0)
+    logging.info(
+        f'Number of non-ref variants for this indvidual: {donor_mt.count()[0]}'
+    )
+
+    # focus on SNVs for now
+    donor_mt = donor_mt.filter_rows(donor_mt.vep.variant_class == 'SNV')
```

Comment:
You could also use Hail's `is_snp` to filter, potentially even before annotating with VEP.

Reply:
Yup!